supervisor/snapshots/snapshot.py

```diff
@@ -370,17 +371,49 @@ def _folder_save(name):
             try:
                 _LOGGER.info("Snapshot folder %s", name)
                 with SecureTarFile(tar_name, "w", key=self._key) as tar_file:
-                    tar_file.add(
-                        origin_dir,
-                        arcname=".",
-                        filter=exclude_filter(MAP_FOLDER_EXCLUDE.get(name, [])),
+                    arcname = "."
+                    tar_file.add(origin_dir, arcname, False)
+                    add_directory_contents(
+                        tar_file, origin_dir, MAP_FOLDER_EXCLUDE.get(name, []), arcname
                     )
 
                 _LOGGER.info("Snapshot folder %s done", name)
                 self._data[ATTR_FOLDERS].append(name)
             except (tarfile.TarError, OSError) as err:
                 _LOGGER.warning("Can't snapshot folder %s: %s", name, err)
 
+        def add_directory_contents(
+            tar_file: tarfile.TarFile, origin_dir, excludes, arcname="."
+        ):
+            """Append directories and/or files to the TarFile if excludes wont filter."""
+
+            def excluded_path(file_path: PurePath):
```

Comment:
Use the exists and well tested function exclude_filter

Reply:
You mean `exclude_filter` from `utils.tar`?
That file requires a `TarInfo` which actually is the main problem, because the `TarInfo` requires access to the I/O where the file is missing...
Or do I miss something?

Reply:
you can make an internal function in tar utils and share it between this two helpers